### DISCUSSION TITLE
50+ Devices - Is this possible?

### DISCUSSION THREAD

**Discussion by agoralive** [Koenkk/zigbee2mqtt#7069] [General]
To be honest, I'm confused....

Some places I read that I can connect 20(21) devices for setup, other places I can connect 100+ as long as I have a bunch of lamps or sockets connected. But is there an answer for my situation:

I have:

8 Xiaomi sensors (battery)
5 Ikea buttons (battery)
1 Ikea USB Extender
1 Philips Hue lamp
2 Ikea 220V sockets
2 Tuya sensors (Battery)

NOW - I would like to add 30+ ZigBee bulbs to my home, what do I need to be able to fulfill this "wish"? I might not stop at these 30+ Bulbs - I might wan't to add an additional 20 later on......

Do I need router(s), Extender(s), plug(s) - What do I need to make sure my zigbee network is working "like a charm" always?

**Comment by m0wlheld** [comment]
> NOW - I would like to add 30+ ZigBee bulbs to my home, what do I need to be able to fulfill this "wish"?

Bulbs. The powered devices in your network will act as router/repeater.

**Comment by agoralive** [comment]
Why will they not pair against the Router?

![image](https://user-images.githubusercontent.com/10817998/114930902-d80b3300-9e35-11eb-96e7-f6fdcd0c59d6.png)

**Comment by Twit123** [comment]
They will find the best/easiest route to follow.

Also the router is showing an LQI of 0, maybe they are avoiding it.

What controller are you using?

Simon

**Reply by agoralive** [reply]
All are the USB stick CC2530, nothing will pair against the Router, ONLY to the coordinator

**Comment by debnetjm** [comment]
@agoralive sure it is possible.
Initially I had CC2531 and connectivity issues started somewhere between 30 and 40 devices.

Now I have CC2652RB and currently 52 devices (30 battery powered) without any issues. 
It is a mix of bulbs, sensors, smart plugs, relays etc from 5 different vendors.

**Comment by c0mplex1** [comment]
Reading the Notes at https://www.zigbee2mqtt.io/information/supported_adapters.html
"Adapters based on the CC2530 or CC2531 chip are not powerful and not recommended for networks larger than 20 devices."
So I suggest using an adapter with a CC2652R, CC2652RB, CC1352P or CC2538.

**Reply by agoralive** [reply]
But I have a router connected which does not allow devices to pair....

I have also tried using HA's zha plugin, using this there is no problems, and devices connect to router without issues. I would just like to get the mqtt possibility which z2m offers

**Comment by kirovilya** [comment]
not my, from here https://forum.iobroker.net/post/615801 : 164 devices on cc2652RB
![image](https://user-images.githubusercontent.com/8360230/115583373-81d53e80-a2d2-11eb-8b27-6bf0a12b0d47.png)
